packages/tracing/src/hubextensions.ts

```diff
@@ -206,15 +208,66 @@ export function startIdleTransaction(
  * @private
  */
 export function _addTracingExtensions(): void {
+  // FIXME: This is problematic, because `getMainCarrier` always return a valid Carrier, but because
+  // of how we set the types, we need to perform an unnecessary check here.
```

Comment:
Is this possible to "fix" right here? Sounds like this comment would be most useful where `getMainCarrier` is defined if the intention is to ever change the signature. Until then, checking for `undefined` at the call sites seems like the way to go.

Reply:
Not sure what's the question here tbh 😅